eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java

```diff
@@ -110,6 +115,37 @@ public ClusteredVariantsRestController(
         }
     }
 
+    @ApiOperation(value = "Find clustered variant (RS) history", notes = "This endpoint returns the history of clustered "
+            + "variants (RS) represented by the given identifier. ")
+    @GetMapping(value = "/{identifier}/history", produces = "application/json")
+    public ResponseEntity<VariantHistory<ClusteredVariant, IClusteredVariant, String, Long>> getVariantHistory(
+            @PathVariable @ApiParam(value = "Numerical identifier of a clustered variant, e.g.: 3000000000",
+                    required = true) Long identifier) throws AccessionDoesNotExistException {
+        List<AccessionResponseDTO<ClusteredVariant, IClusteredVariant, String, Long>> allVariants =
+                new ArrayList<>();
+        try {
+            allVariants.addAll(getNonHumanClusteredVariants(identifier));
+            allVariants.addAll(humanService.getAllByAccession(identifier).stream().map(this::toDTO)
+                    .collect(Collectors.toList()));
+        } catch (AccessionDeprecatedException | AccessionMergedException e) {
+
+        }
+
+        List<HistoryEventDTO<Long, ClusteredVariant>> allOperations = clusteredVariantOperationService.getAllOperations(identifier)
+                .stream().map(this::toHistoryEventDTO).collect(Collectors.toList());
+
+        if (allVariants.isEmpty() && allOperations.isEmpty()) {
+            throw new AccessionDoesNotExistException(identifier);
+        }
+        return ResponseEntity.ok(new VariantHistory<>(allVariants, allOperations));
+    }
+
+    public HistoryEventDTO<Long, ClusteredVariant> toHistoryEventDTO(IEvent<? extends IClusteredVariant, Long> operation) {
+        HistoryEvent<IClusteredVariant, Long> historyEvent = new HistoryEvent<IClusteredVariant, Long>(operation.getEventType(), operation.getAccession(),
+                operation.getInactiveObjects().get(0).getVersion(), operation.getResultInto(), operation.getCreatedDate(), operation.getInactiveObjects().get(0).getModel());
```

Comment:
Please mind the lines length, it makes it easier to review it they don't exceed 120 characters

Reply:
sorry, looks like formatting issue again. Will correct